From 281088544b0a710ddd9a945cf17b25fbfa4f2023 Mon Sep 17 00:00:00 2001
From: Mathieu Bridon <bochecha@daitauha.fr>
Date: Fri, 17 Aug 2018 21:32:16 +0200
Subject: [PATCH 22/27] python: difflib prefers unicode strings

Python 3 does not automatically convert from bytes to unicode strings
like Python 2 used to do.

This commit makes sure we pass unicode strings to difflib.unified_diff,
so that the script works on both Python 2 and 3.

Reviewed-by: Dylan Baker <dylan@pnwbakers.com>
---
 src/compiler/glsl/glcpp/tests/glcpp_test.py | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/src/compiler/glsl/glcpp/tests/glcpp_test.py b/src/compiler/glsl/glcpp/tests/glcpp_test.py
index 8c7552124a..8c613cdf6e 100644
--- a/src/compiler/glsl/glcpp/tests/glcpp_test.py
+++ b/src/compiler/glsl/glcpp/tests/glcpp_test.py
@@ -63,8 +63,9 @@ def test_output(glcpp, filename, expfile, nl_format='\n'):
             stderr=subprocess.STDOUT,
             stdin=subprocess.PIPE)
         actual, _ = proc.communicate(f.read())
+        actual = actual.decode('utf-8')
 
-    with open(expfile, 'rb') as f:
+    with open(expfile, 'r') as f:
         expected = f.read()
 
     if actual == expected:
-- 
2.18.1

